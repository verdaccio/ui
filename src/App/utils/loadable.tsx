import React, { lazy, Suspense } from 'react';
export default (importCallback: any) => {
  const TargetComponent = lazy(importCallback);
  return (props: any) => (
    <Suspense fallback={null}>
      <TargetComponent {...props} />
    </Suspense>
  );
};
