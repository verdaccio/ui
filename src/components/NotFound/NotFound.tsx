import styled from '@emotion/styled';
import React, { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import Box from '../Box';
import Button from '../Button';
import Heading from '../Heading';

import PackageImg from './img/package.svg';

const NotFound: React.FC = () => {
  const history = useHistory();
  const { t } = useTranslation();

  const handleGoHome = useCallback(() => {
    history.push('/');
  }, [history]);

  return (
    <Box
      alignItems="center"
      data-testid="404"
      display="flex"
      flexDirection="column"
      flexGrow={1}
      justifyContent="center"
      p={2}>
      <EmptyPackage alt={t('error.404.page-not-found')} src={PackageImg} />
      <StyledHeading className="not-found-text" variant="h4">
        {t('error.404.sorry-we-could-not-find-it')}
      </StyledHeading>
      <Button data-testid="not-found-go-to-home-button" onClick={handleGoHome} variant="contained">
        {t('button.go-to-the-home-page')}
      </Button>
    </Box>
  );
};

export default NotFound;

const EmptyPackage = styled('img')({
  width: '150px',
  margin: '0 auto',
});

const StyledHeading = styled(Heading)<{ theme?: Theme }>(({ theme }) => ({
  color: theme?.palette.type === 'light' ? theme?.palette.primary.main : theme?.palette.white,
  marginBottom: 16,
}));
