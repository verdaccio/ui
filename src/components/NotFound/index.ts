export { default, NOT_FOUND_TEXT } from './NotFound';
