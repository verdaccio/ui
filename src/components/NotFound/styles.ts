import styled from '@emotion/styled';

import { default as MuiCard } from '../Card';
import { default as Typography } from '../Heading';
import List from '../List';

export const Wrapper = styled('div')({
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  justifyContent: 'center',
  flex: 1,
  padding: '16px',
});

export const Inner = styled('div')({
  maxWidth: '650px',
  display: 'flex',
  flexDirection: 'column',
});

export const EmptyPackage = styled('img')({
  width: '150px',
  margin: '0 auto',
});

export const Heading = styled(Typography)({
  color: '#4b5e40',
});

export const StyledList = styled(List)({
  padding: 0,
  color: '#4b5e40',
});

export const Card = styled(MuiCard)({
  marginTop: '24px',
});
