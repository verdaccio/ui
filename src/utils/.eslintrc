{
    "rules": {
        "no-invalid-this": 0
    }
}