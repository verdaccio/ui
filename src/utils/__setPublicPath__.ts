// @ts-ignore
if (!__DEBUG__) {
  // @ts-ignore
  __webpack_public_path__ = window.VERDACCIO_API_URL.replace(/\/verdaccio\/$/, '/static/'); // eslint-disable-line
}
