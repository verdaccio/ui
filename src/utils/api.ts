import storage from './storage';
import '../../types';

/**
 * Handles response according to content type
 * @param {object} response
 * @returns {promise}
 */
export function handleResponseType(response: Response): Promise<[boolean, any]> {
  if (response.headers) {
    const contentType = response.headers.get('Content-Type');
    if (contentType && contentType.includes('application/pdf')) {
      return Promise.all([response.ok, response.blob()]);
    }
    if (contentType && contentType.includes('application/json')) {
      return Promise.all([response.ok, response.json()]);
    }
    // it includes all text types
    if (contentType && contentType.includes('text/')) {
      return Promise.all([response.ok, response.text()]);
    }

    // unfortunately on download files there is no header available
    if (response.url && response.url.endsWith('.tgz') === true) {
      return Promise.all([response.ok, response.blob()]);
    }
  }

  return Promise.all([response.ok, response.text()]);
}

class API {
  public request<T>(url: string, method = 'GET', options: RequestInit = { headers: {} }): Promise<T> {
    if (!window.VERDACCIO_API_URL) {
      throw new Error('VERDACCIO_API_URL is not defined!');
    }

    const token = storage.getItem('token');
    const headers = new Headers(options.headers);

    if (token && headers.has('Authorization') === false) {
      headers.set('Authorization', `Bearer ${token}`);
      options.headers = headers;
    }

    if (!['http://', 'https://', '//'].some(prefix => url.startsWith(prefix))) {
      url = window.VERDACCIO_API_URL + url;
    }

    return new Promise((resolve, reject) => {
      fetch(url, {
        method,
        credentials: 'same-origin',
        ...options,
      })
        .then(handleResponseType)
        .then(response => {
          if (response[0]) {
            resolve(response[1]);
          } else {
            reject(new Error('something went wrong'));
          }
        })
        .catch(error => {
          reject(error);
        });
    });
  }
}

export default new API();
