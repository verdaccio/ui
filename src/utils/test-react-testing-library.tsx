import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';

import ThemeProvider from 'verdaccio-ui/design-tokens/ThemeProvider';
import APIProvider from 'verdaccio-ui/providers/API/APIProvider';
import AppConfigurationProvider from 'verdaccio-ui/providers/config';

import i18nConfig from '../../i18n/config';

const customRender = (node: React.ReactElement, ...options: any) => {
  return render(
    <AppConfigurationProvider>
      <APIProvider>
        <ThemeProvider>
          <I18nextProvider i18n={i18nConfig}>{node}</I18nextProvider>
        </ThemeProvider>
      </APIProvider>
    </AppConfigurationProvider>,
    ...options
  );
};

export * from '@testing-library/react';
export { customRender as render };
