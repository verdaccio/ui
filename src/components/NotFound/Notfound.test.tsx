import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { render, fireEvent } from 'verdaccio-ui/utils/test-react-testing-library';

import NotFound from './NotFound';

describe('<NotFound /> component', () => {
  test('should load the component in default state', () => {
    const { container } = render(
      <Router>
        <NotFound />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('go to Home Page button click', async () => {
    const spy = jest.spyOn(React, 'useCallback');
    const { getByTestId } = render(
      <Router>
        <NotFound />
      </Router>
    );

    const node = getByTestId('not-found-go-to-home-button');
    fireEvent.click(node);
    expect(spy).toHaveBeenCalled();
  });
});
