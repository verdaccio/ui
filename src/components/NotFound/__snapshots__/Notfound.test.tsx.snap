// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> component should load the component in default state 1`] = `
.emotion-0 {
  width: 150px;
  margin: 0 auto;
}

.emotion-2 {
  color: #4b5e40;
  margin-bottom: 16px;
}

<div
  class="MuiBox-root MuiBox-root-1"
  data-testid="404"
>
  <img
    alt="404 - Page not found"
    class="emotion-0 emotion-1"
    src="[object Object]"
  />
  <h4
    class="MuiTypography-root not-found-text emotion-2 emotion-3 MuiTypography-h4"
  >
    Sorry, we couldn't find it...
  </h4>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained"
    data-testid="not-found-go-to-home-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Go to the home page
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
